File: src/main/java/org/intranet/entity/News.java
package org.intranet.entity;

import com.fasterxml.jackson.annotation.JsonFormat;

import javax.persistence.Column;
import javax.persistence.Entity;
import javax.persistence.PrePersist;
import java.time.LocalDateTime;

@Entity
public class News extends ParentEntity {
    private String title;
    @Column(columnDefinition = "TEXT")
    private String text;
    @JsonFormat(pattern="yyyy-MM-dd hh:mm")
    private LocalDateTime date;

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getText() {
        return text;
    }

    public void setText(String text) {
        this.text = text;
    }

    public LocalDateTime getDate() {
        return date;
    }

    public void setDate(LocalDateTime date) {
        this.date = date;
    }

    @PrePersist
    void date() {
        this.date = LocalDateTime.now();
    }
}

File: src/main/java/org/intranet/entity/News.kt
package org.intranet.entity

import com.fasterxml.jackson.annotation.JsonFormat
import java.time.LocalDateTime
import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.PrePersist

@Entity
class News : ParentEntity() {
    var title: String? = null
    @Column(columnDefinition = "TEXT")
    var text: String? = null
    @JsonFormat(pattern = "yyyy-MM-dd hh:mm")
    var date: LocalDateTime? = null

    @PrePersist
    internal fun date() {
        this.date = LocalDateTime.now()
    }
}
